### ISSUE TITLE
Model Computed Properties

### ISSUE THREAD

**Issue by thec0keman** [miragejs/miragejs#640]
It would be really nice to define computed attributes that would be calculated at serialization time.  This can be useful in situations where a backend would return generated data structures or denormalized data.

Based on the design of mirage, I think a simple way to add this would be to have a `computedAttrs` attribute on the serializer.   Then, in the [_attrsForModel](https://github.com/miragejs/miragejs/blob/master/lib/serializer.js#L733) could have an additional step where it iterates on the `computedAttrs` key on the serializer and calls getters on the model for those values.

So as an example:
```
const CommentSerializer = RestSerializer.extend({
  computedAttrs: ['authorName']
});

const CommentModel = Model.extend({
  author: belongsTo(),

  authorName() {
    if (this.author) {
      return this.author.name;
    }
  }
});
```

**Comment by samselikoff** [comment]
Yes, I think we had a PR for this at one point. Synthetic attributes. I am 100% in favor of this if someone wants to work on it.

**Comment by samselikoff** [comment]
https://github.com/miragejs/discuss/issues/6
